runner/entrypoint.sh

```diff
@@ -44,70 +64,91 @@ if [ -z "${RUNNER_REPO}" ] && [ -n "${RUNNER_GROUP}" ];then
 fi
 
 # Hack due to https://github.com/actions-runner-controller/actions-runner-controller/issues/252#issuecomment-758338483
-if [ ! -d /runner ]; then
-  echo "/runner should be an emptyDir mount. Please fix the pod spec." 1>&2
+if [ ! -d "${RUNNER_HOME}" ]; then
+  error "${RUNNER_HOME} should be an emptyDir mount. Please fix the pod spec."
   exit 1
 fi
 
-sudo chown -R runner:docker /runner
-cp -r /runnertmp/* /runner/
+# if this is not a testing environment
+if [ -z "${UNITTEST:-}" ]; then
+  sudo chown -R runner:docker ${RUNNER_HOME}
+  mv /runnertmp/* ${RUNNER_HOME}/
+fi
```

Comment:
@sledigabel Hey! Thanks for all your efforts. I like the general direction we're heading in this PR.

And this part (L73-76) is the only blocker before merging for me.

Can we somehow remove this unit-test-only conditional? I generally prefer not to have test code leaking into the implementation.

My quick idea would be to inject all the dependencies (the user and the group to be chown-ed, the directory which contains initial data for the RUNNER_HOME dir):

```suggestion
sudo chown -R ${RUNNER_USER}:${DOCKER_USER} ${RUNNER_HOME}
mv ${RUNNER_INIT}/* ${RUNNER_HOME}/
fi
```

You'd also need to have defaults for these new variables, perhaps it would look like:

```
RUNNER_USER=${RUNNER_USER:-runner}
DOCKER_USER=${DOCKER_USER:-docker}
RUNNER_INIT=${RUNNER_INIT:-/runnertmp}
```